PYTHON/IrisUtils/hdf5_lib.py

```diff
@@ -896,27 +895,53 @@ def load_tx_data(metadata, dirpath):
         zero_sc_ind = np.setdiff1d(zero_sc_ind, pilot_sc_ind)
         nonzero_sc_ind = np.setdiff1d(range(fft_size), zero_sc_ind)
         ul_equal_syms_frame_num = int(metadata['UL_DATA_FRAME_NUM'])
-        tx_file_names = metadata['TX_FD_DATA_FILENAMES'].astype(str)
+        #
+        # aaa = metadata['TX_FD_DATA_FILENAMES']
+        # print(type(aaa))
+        # aaa = aaa.astype(str)
+        # print(type(aaa))
+        #
+        # manually_file = ['ul_data_f_16QAM_52_64_10_1_1_AB_0.bin']
+        # manually_file = ['ul_data_f_QPSK_52_64_10_1_1_AB_0.bin']
+        manually_file = ['ul_data_f_64QAM_52_64_10_1_1_AB_0.bin']
```

Comment:
why is this manually set?

Reply:
cause somehow in the saved hdf5 file, for uplink transmissions, the `TX_FD_DATA_FILENAMES` section is missing. So for the testing purposes, I just set it manually.  An example output can be seen [https://docs.google.com/presentation/d/1AFXKeAOm8HbT-o-kMyGJTGtRnMOKC2cAyBhUI0VPuAs/edit?usp=sharing](url)

Reply:
`TX_FD_DATA_FILENAMES` always gets recorded as long as there are uplink files. This feature has been added a long time ago and is mainlined in the `master` branch too. Most likely, the dataset you are showing in the slides was collected using a very old version. It could be also a corner case of the configs. Please test the latest develop branch again with a known good json and your own json, and let me know if you see this bad behavior

Having said that, manually setting the file name is obviously bad and should not be done. It also broke my tests. So please revert to the original way which is reading the file from the HDF5 dataset. If there is an issue with the Sounder not recording the filename in the metadata, we should fix that instead.

Reply:
Thanks for providing the detailed explanation, will test and look through it again

Reply:
I figured out where the issue mentioned in the previous slides comes from. 

It comes up when I use `--bs_only` and `--client_only` flags when running sounder; in this case, the BS will have no reference to where the TX_FD_DATA_FILENAMES the client is referring to and will not include the `TX_FD_DATA_FILENAMES`

Reply:
I just converted the post-processing python files I changed back to the original develop version.